### DISCUSSION TITLE
Wish의 수량에 대한 테이블 수정 방식

### DISCUSSION THREAD

**Discussion by YeaChan05** [KakaoFunding/back-end#31] [Ideas]
영상 보시면 인스타그램에서는 like를 게시글 테이블에 수량만 넣어두는 방법을 채택하고 있네요
저도 지금 상품을 카테고리 기준으로 조회하는 로직의 테스트 케이스를 작성하고 있는데 `Wish`를 생성하지 않은 상태에서 테스트코드를 작성해보니 `Wish`수 기준으로 정렬을 하려 할때 모두 0개이기 때문에 정렬이 기본 정렬 기준인 상품 이름 기준으로 정렬되더라고요

추후에 `Wish` 정보를 `@OneToMany`로 양방향 매핑하는 방법 대신 비즈니스 로직으로 풀어나가는 방법은 어떨까요?
https://youtu.be/V27XkmVPqYQ?t=323&feature=shared

**Comment by kmw2378** [comment]
어제 봤던 영상인데 예찬님도 보셨군요 ㅋㅋㅋ 신기합니다! 영상에서 유저마다 좋아요한 피드를 보여주기 위해 Like 테이블을 사용하듯 저희도 회원마다 Wish 상품을 보여줘야 하므로 Wish 테이블을 필수일 것 같습니다. 다만 상품별 조회 횟수는 `Product` 테이블에 별도의 컬럼을 두는 방법이 좋을 것 같아요.

양방향 대신 비지니스 로직으로 풀어나가는거에 동의합니다. `WishRepository`, `ProductRepository` 를 필드로 하는 클래스를 둬서 해결하는 방법도 괜찮을 것 같아요!

**Reply by YeaChan05** [reply]
추후에 성능 개선과정에서 조정해보도록 하겠습니다!
